### DISCUSSION TITLE
Icon color doesn't invert with theme color change

### DISCUSSION THREAD

**Discussion by david-pa** [microsoft/vscode-discussions#526] [Extension Development QnA] [answered]
I am working on a VSCode extension that uses icons when hovering over items in a tree view. I believe the icons came from the [official VSCode icon library](https://github.com/microsoft/vscode-icons), and I do have the dark and light versions of them, but when an item is selected in Light Mode, they're very hard to make out.

<img width="154" alt="Screenshot 2023-03-15 at 4 39 30 PM" src="https://user-images.githubusercontent.com/108750899/225449902-20d9a7c9-69f0-4e36-acc2-5e1bd537f88d.png">

I think they are supposed to be inverted to white automatically per the theme, like the text is, but that doesn't seem to  happening. Is there something that needs to be done within my extension or within the svg file for the icons to enable that behavior?

**Comment by alefragnani** [comment, marked as answer]
Hi @david-pa ,

If you are using VS Code native/built-in icons, I suggest you to use Codicons (https://microsoft.github.io/vscode-codicons/dist/codicon.html) instead. If you use Codicons, you not only won’t have to worry about `light`/`dark` anymore, but your icons will also embrace `workbench.colorCustomizations` colors as well. And of course, your extension won’t have to bundle that SVG anymore.

Its declaration would be something like this:

```json
        {
                "command": "_projectManager.openInNewWindow",
                "title": "%projectManager.commands.openInNewWindow.title%",
                "icon": "$(link-external)" // <<— here
        },

```

On the other hand, if you really need an SVG from that library, be sure you defined both `dark` and `light` options for the command. If you did, maybe the icon you copied had some issue, so I suggest you to report on that repo about it. I have used that library a lot at the beginning, but the only issue I had was about some versions (size) not being too sharp.

Hope this helps.

**Reply by david-pa** [reply]
@alefragnani  Is there anything specific that would have to be done to a custom SVG (a completely original icon) for it to work with `workbench.colorCustomizations` settings? I'm guessing that's what controls the inversion of colors and things like that when switching between UI color themes? Please correct me if I'm wrong.

**Reply by alefragnani** [reply]
I remember walkthrough SVGs are the themeable (https://github.com/microsoft/vscode/issues/130153), but I’m not sure _every SVG_ is. You could try out the same approach on yours.

Otherwise, you simply use different icons for light/dark, like https://code.visualstudio.com/api/references/contribution-points#command-example

```json
"commands": [
      {
        "command": "extension.sayHello",
        "title": "Hello World",
        "category": "Hello",
        "icon": {
          "light": "path/to/light/icon.svg",
          "dark": "path/to/dark/icon.svg"
        }
      }
```
